action/messageHubFeed.js

```diff
@@ -10,36 +10,26 @@ var request = require('request');
  *  @param {string} endpoint - address to OpenWhisk deployment
  */
 function main(params) {
-    if(!params.package_endpoint) {
-        whisk.error('Could not find the package_endpoint parameter.');
-        return;
-    }
-
-    var triggerComponents = params.triggerName.split("/");
-    var namespace = encodeURIComponent(process.env['__OW_NAMESPACE']);
-    var trigger = encodeURIComponent(triggerComponents[2]);
-    var feedServiceURL = 'http://' + params.package_endpoint + '/triggers/' + namespace + '/' + trigger;
-
-    if (params.lifecycleEvent === 'CREATE') {
-        var validatedParams = validateParameters(params);
-        if (!validatedParams) {
-            // whisk.error has already been called.
-            // all that remains is to bail out.
+    var promise = new Promise(function(resolve, reject) {
+        if(!params.package_endpoint) {
+            reject('Could not find the package_endpoint parameter.');
             return;
         }
 
-        // make sure we have valid MH credentials
-        console.log("Checking Message Hub credentials...");
-        return checkMessageHubCredentials(validatedParams)
-            .then(function() {
-                console.log("Successfully authenticated with Message Hub");
+        var triggerComponents = params.triggerName.split("/");
+        var namespace = encodeURIComponent(process.env['__OW_NAMESPACE']);
+        var trigger = encodeURIComponent(triggerComponents[2]);
+        var feedServiceURL = 'http://' + params.package_endpoint + '/triggers/' + namespace + '/' + trigger;
 
+        if (params.lifecycleEvent === 'CREATE') {
+            // makes a PUT request to create the trigger in the feed service provider
+            var putTrigger = function(validatedParams) {
```

Comment:
I kind of agree. Inner functions are weird and confusing.

That being said, this function isn't used anywhere else _and_ it captures the `triggerURL` and `namespace` vars declared above.